docs/concepts/feature-generation.md

```diff
@@ -1,16 +1,19 @@
 ---
 layout: default
-title: Feathr Feature Generation
+title: Feature Generation and Materialization
 parent: Feathr Concepts
 ---
 
-# Feature Generation
-Feature generation is the process to create features from raw source data into a certain persisted storage.
+# Feature Generation and Materialization
 
-User could utilize feature generation to pre-compute and materialize pre-defined features to online and/or offline storage. This is desirable when the feature transformation is computation intensive or when the features can be reused(usually in offline setting). Feature generation is also useful in generating embedding features. Embedding distill information from large data and it is usually more compact.
+Feature generation (also known as feature materialization) is the process to create features from raw source data into a certain persisted storage in either offline store (for further reuse), or online store (for online inference).
+
+User could utilize feature generation to pre-compute and materialize pre-defined features to online and/or offline storage. This is desirable when the feature transformation is computation intensive or when the features can be reused (usually in offline setting). Feature generation is also useful in generating embedding features, where those embeddings distill information from large data and is usually more compact.
 
 ## Generating Features to Online Store
-When we need to serve the models online, we also need to serve the features online. We provide APIs to generate features to online storage for future consumption. For example:
+
+When we need to serve the models online, we also need to serve the features online. Feathr provides APIs to generate features to online storage for future consumption. For example:
```

Comment:
>When we need to serve the models online, we also need to serve the features online. [](http://example.com/codeflow?start=0&length=83)

When we need to serve the models online, we also need to serve the features online.

Contents are duplicated